libbeat/esleg/eslegclient/connection.go

```diff
@@ -326,7 +326,6 @@ func (conn *Connection) Ping() (ESPingData, error) {
 // Close closes a connection.
 func (conn *Connection) Close() error {
 	conn.HTTP.CloseIdleConnections()
-	conn.cancelReqs()
```

Comment:
Looking at the diff from https://github.com/elastic/beats/pull/40572, which introduced the issue, it seems that `cancelReqs` is only called here, and you ended up removing it as the fix. Does that mean we should just remove the other references to it as well?

Reply:
That's a quick fix to fix our released branches while we work on the actual fix, hence I only removed the line causing the issue and left the rest of the feature. Soon I'll make a PR with tests and a proper fix.